test/e2e/k8s_util.go

```diff
@@ -100,39 +100,62 @@ func (k *KubernetesUtils) Probe(ns1, pod1, ns2, pod2 string, port int32, protoco
 		return Error, fmt.Errorf("no Pod of label pod=%s in Namespace %s found", pod2, ns2)
 	}
 	toPod := toPods[0]
-	toIP := toPod.Status.PodIP
-
-	// There seems to be an issue when running Antrea in Kind where tunnel traffic is dropped at
-	// first. This leads to the first test being run consistently failing. To avoid this issue
-	// until it is resolved, we try to connect 3 times.
-	// See https://github.com/vmware-tanzu/antrea/issues/467.
-	cmd := []string{
-		"/bin/sh",
-		"-c",
-	}
-	switch protocol {
-	case v1.ProtocolTCP:
-		cmd = append(cmd, fmt.Sprintf("for i in $(seq 1 3); do /agnhost connect %s:%d --timeout=1s --protocol=tcp && exit 0 || true; done; exit 1", toIP, port))
-	case v1.ProtocolUDP:
-		cmd = append(cmd, fmt.Sprintf("for i in $(seq 1 3); do /agnhost connect %s:%d --timeout=1s --protocol=udp && exit 0 || true; done; exit 1", toIP, port))
-	case v1.ProtocolSCTP:
-		cmd = append(cmd, fmt.Sprintf("for i in $(seq 1 3); do /agnhost connect %s:%d --timeout=1s --protocol=sctp && exit 0 || true; done; exit 1", toIP, port))
-	}
-	// HACK: inferring container name as c80, c81, etc, for simplicity.
-	containerName := fmt.Sprintf("c%v", port)
-	log.Tracef("Running: kubectl exec %s -c %s -n %s -- %s", fromPod.Name, containerName, fromPod.Namespace, strings.Join(cmd, " "))
-	stdout, stderr, err := k.runCommandFromPod(fromPod.Namespace, fromPod.Name, containerName, cmd)
-	if err != nil {
-		// log this error as trace since may be an expected failure
-		log.Tracef("%s/%s -> %s/%s: error when running command: err - %v /// stdout - %s /// stderr - %s", ns1, pod1, ns2, pod2, err, stdout, stderr)
-		// do not return an error
-		if strings.Contains(stderr, "TIMEOUT") {
-			return Dropped, nil
+
+	// Both IPv4 and IPv6 address should be tested.
+	toIPs := toPod.Status.PodIPs
+	if len(toIPs) == 0 && toPod.Status.PodIP != ""{
+		toIPs = append(toIPs, v1.PodIP{IP: toPod.Status.PodIP})
+	}
```

Comment:
is there a reason why we don't use the information from the `podIPs` map here?

More questions about the existing code:
* what is the `toPod.Status.PodIP != ""` check for?
* why do we need to use the `v1.PodIP` type to build our slice of IPs?

Reply:
I thought in this way, we can use this `Probe` func when `podIPs`  is not initialized. So `Probe` func doesn't have a prerequisite initialization in antreapolicy test.
About `toPod.Status.PodIP != ""`, the discussion with @Dyanngg is here: https://github.com/vmware-tanzu/antrea/pull/2001#discussion_r603670277. Just to avoid if we are exec probe when a Pod hasn't been allocated an IP then insert "" to the slice.
About `v1.PodIP`, it's just because `toPod.Status.PodIPs` is a []v1.PodIP.

Reply:
Addressed your last two questions.